scripts/deployDomainRegistry.js

```diff
@@ -0,0 +1,33 @@
+// We require the Hardhat Runtime Environment explicitly here. This is optional
+// but useful for running the script in a standalone fashion through `node <script>`.
+//
+// You can also run a script with `npx hardhat run <script>`. If you do that, Hardhat
+// will compile your contracts, add the Hardhat Runtime Environment's members to the
+// global scope, and execute the script.
+const hre = require("hardhat");
+
+async function main() {
+  const currentTimestampInSeconds = Math.round(Date.now() / 1000);
+  const unlockTime = currentTimestampInSeconds + 60;
+
+  const lockedAmount = hre.ethers.parseEther("0.001");
+
+  const lock = await hre.ethers.deployContract("DomainRegistry", [unlockTime], {
+    value: lockedAmount,
+  });
```

Comment:
You didn't clean up the code from the default Hardhat example. Besides that, you put some ether to the contract when you deploy. What is the reason? And why the contract `constructor` is payable?   -1 point